### DISCUSSION TITLE
Has the hubitat maker api improved in recent years?

### DISCUSSION THREAD

**Discussion by evelant** [jason0x43/hacs-hubitat#286] [General]
Several years ago I used this addon to keep my hubitat as my primary zigbee/zwave hub while controlling everything through home assistant. That didn't work well because at the time the hubitat maker api was quite buggy, poorly documented, and quite limited. It would often fail on me resulting in devices being out of sync, not responding, or just being slow. When I asked about this on the hubitat forum a staff member more or less told me "why don't you write a better one?" (I would if I had the time!) which I took to mean they have no interest in providing a robust "maker integration". However, that was on an old hubitat hub and it's been several years so I'm curious:

 Has hubitat's API gotten any better? How reliable and responsive is it on a newer hub like c8-pro? 

I ask this because I'm becoming quite frustrated with the bugginess in both zigbee2mqtt and zha. My wife is constantly irritated by broken automations and non-responding devices. One thing I remember hubitat being good at was reliably communicating with zigbee devices moreso than zha or z2m. If hubitat can be used as a reliable zigbee gateway I might want to pick one up again. Thanks for any insight!

**Comment by jason0x43** [comment]
I used to use Hubitat as a zigbee and zwave hub with the Maker API, and it was fine. My main issues with it were its opaqueness and its sometimes weak device support. The Maker API has always worked well enough, though. It’s very responsive, and it’s fine for accessing the core functionality of devices.

Home Assistant, in my experience, tends to support a wider range of devices than Hubitat, and new devices are supported more quickly. It also provides access to more device features and makes diagnosing issues easier because it’s open source.

I’ve been using HA with a SkyConnect and ZHA for the last few years for my zigbee devices. The only issue I’ve run into has been when my wifi access point has decided to pick a channel that overlapped with my zigbee channel. That’s happened a couple of times, and it causes a lot of random flakiness in the zigbee network. One of the nice things about HA is that I can look in the logs and see ZHA telling me that my zigbee channel is 99% occupied, and maybe I should do something about that. 🙂
